spec/integration/errors_spec.rb

```diff
@@ -3,20 +3,21 @@
 
 describe "When errors occur" do
 
-  def set_consider_all_requests_local(value)
-    method = Rails.application.method(:env_config)
-    allow(Rails.application).to receive(:env_config).with(no_args) do
-      method.call.merge(
-        'action_dispatch.show_exceptions' => true,
-        'consider_all_requests_local' => value
-      )
-    end
-    @requests_local = Rails.application.config.consider_all_requests_local
-    Rails.application.config.consider_all_requests_local = value
-  end
+  def set_consider_all_requests_local(value, example)
```

Comment:
From the commit message, I don't really understand _why_ this change fixes the problem.

Presumably something to do with setting the values properly rather than stubbing them?

Reply:
I've split this commit into 1. the fix 13f9b7df81 where I've improved the commit message, 2. refactorings 0503924607, f75db140e2 and 70e8d58ab5.